pkg/comp-functions/runtime/function_mgr.go

```diff
@@ -1064,3 +1069,45 @@ func (s *ServiceRuntime) addUsages() error {
 	}
 	return nil
 }
+
+func (s *ServiceRuntime) ForwardEvents() error {
+	claimRef := s.observedComposite.GetClaimReference()
```

Comment:
We probably drop some events before the composite reference is available though.

Reply:
No they are not, and I don't even care. I had the same problem with desired, some times I don't have anything in there. Do you still want to go with desired?

Reply:
I guess it's fine for now.